app/uk/gov/hmrc/gform/form/FormController.scala

```diff
@@ -139,6 +139,19 @@ class FormController(
     formService.getKeyStore(formId).asOkJson
   }
 
+  def enrolmentCallBack(formId: FormId): Action[AnyContent] = Action.async { implicit request =>
+    val audit = new Connector {}
+    val requestHandler = new RequestConverter
+    val event = requestHandler.requestToDataEvent(formId, request)
+
+    audit
+      .sendRequest(event)
+      .recover {
+        case _: Exception => Results.NotFound
+      }
+      .map(_ => Results.Ok)
```

Comment:
```suggestion
      
  def enrolmentCallBack(formId: FormId): Action[AnyContent] = Action.async { implicit request =>
    val requestHandler = RequestConverter
    val event = requestHandler.requestToDataEvent(formId, request)

    auditConnector
      .sendRequest(event)
      .collect {
        case _: Exception => Results.NotFound
        case _            => Results.Ok
      }
  }
```

how about this? @josef-vlach

Reply:
@PasGat

Reply:
This is not correct. `case` branch on line 160 will never be hit. Since `Future`s are omnipresent on the platform their understanding is vital. There is nice article on scala site about them:
https://docs.scala-lang.org/overviews/core/futures.html

Reply:
@josef-vlach How about this? from some manual testing and hardcoding this seems to work


  def enrolmentCallBack(formId: FormId): Action[AnyContent] = Action.async { implicit request =>
    val requestHandler = RequestConverter
    val event = requestHandler.requestToDataEvent(formId, request)

    auditConnector
      .sendRequest(event)
      .recover {
        case _: Exception => Results.NotFound
      }
      .map {
        case Results.NotFound => Results.NotFound
        case _                => Results.Ok
      }
  }

Reply:
***correction 


  def enrolmentCallBack(formId: FormId): Action[AnyContent] = Action.async { implicit request =>
    val requestHandler = RequestConverter
    val event = requestHandler.requestToDataEvent(formId, request)

    auditConnector
      .sendRequest(event)
      .recover {
        case _: Exception => AuditResult.Failure

      }
      .map {
        case AuditResult.Failure => Results.NotFound
        case _                => Results.Ok
      }
  }